car-service/src/main/java/xws/team16/carservice/model/Car.java

```diff
@@ -72,4 +72,19 @@
    @OneToMany(mappedBy = "car")
    private Set<Ad> ads;
 
+   public Float getAverageGrade(){
```

Comment:
I think this function shouldn't be in model class, but rather service class. Model should be POJO with only constructor, getters and setters.